bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestSnapshots.java

```diff
@@ -80,53 +160,85 @@ protected void cloneSnapshot(String snapshotName, TableName tableName)
   }
 
   @Override
-  protected void deleteSnapshots(Pattern pattern) throws IOException {
+  protected int listSnapshotsSize(Pattern pattern) throws IOException {
     try (Admin admin = getConnection().getAdmin()) {
-      admin.deleteSnapshots(pattern);
+      return admin.listSnapshots(pattern).size();
     }
   }
 
   @Override
-  protected void deleteTableSnapshots(Pattern tableName, Pattern snapshotName) throws IOException {
+  protected void deleteTable(TableName tableName) throws IOException {
     try (Admin admin = getConnection().getAdmin()) {
-      admin.deleteTableSnapshots(tableName, snapshotName);
+      admin.deleteTable(tableName);
     }
   }
 
-  @Override
-  protected int listTableSnapshotsSize(String tableNameRegex, String snapshotNameRegex)
-      throws IOException {
+  protected Map<String, Long> createAndPopulateTable(TableName tableName) throws IOException {
+    Map<String, Long> values = new HashMap<>();
     try (Admin admin = getConnection().getAdmin()) {
-      return admin.listTableSnapshots(tableNameRegex, snapshotNameRegex).size();
-    }
-  }
+      HTableDescriptor descriptor = new HTableDescriptor(tableName);
+      descriptor.addFamily(new HColumnDescriptor(COLUMN_FAMILY));
+      admin.createTable(descriptor);
 
-  @Override
-  protected int listSnapshotsSize(Pattern pattern) throws IOException {
-    try (Admin admin = getConnection().getAdmin()) {
-      return admin.listSnapshots(pattern).size();
+      try (Table table = getConnection().getTable(tableName)) {
+        List<Put> puts = new ArrayList<>();
+        for (long i = 0; i < 10; i++) {
+          final UUID rowKey = UUID.randomUUID();
+          byte[] row = Bytes.toBytes(rowKey.toString());
+          values.put(rowKey.toString(), i);
+          puts.add(new Put(row).addColumn(COLUMN_FAMILY, QUALIFIER, Bytes.toBytes(i)));
+        }
+        table.put(puts);
+      }
     }
+    return values;
   }
 
-  @Override
-  protected int listTableSnapshotsSize(Pattern tableNamePattern, Pattern snapshotNamePattern)
-      throws IOException {
-    try (Admin admin = getConnection().getAdmin()) {
-      return admin.listTableSnapshots(tableNamePattern, snapshotNamePattern).size();
+  protected void createBackupAndWait(Admin admin, String backupId, TableName tableName)
```

Comment:
here and everywhere else, why wait?

Reply:
If we do not wait, the next assertion (usually a `listSnapshots` call) could fail as it could be executed too quickly. The same was done in the veneer tests: https://github.com/googleapis/java-bigtable/blob/master/google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/admin/v2/it/BigtableBackupIT.java#L341

Reply:
nvm, after retesting this was no longer necessary. removing